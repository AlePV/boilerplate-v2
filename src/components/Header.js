import React from "react";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {startLogout} from "../actions/auth";

export const Header = ({startLogout}) => (
    <header className="header">
        <div className="content-container">
            <div className="header__container">
                <Link className="header__title" to="/dashboard">
                <h1>Boilerplate</h1>
                </Link>
                <button  className="login-button login-button--link" onClick={startLogout}>Logout</button>
            </div>
        </div>
    </header>
);

// for the bold letters --> shows where you are
const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);

// Create private state for user data --> Private Firebase Data
// '--> Change the firebase rules
// '--> user/userUid/expenses/idOfExpense