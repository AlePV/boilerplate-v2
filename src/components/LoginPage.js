import React from "react";
import {connect} from "react-redux";
import {startLogin} from "../actions/auth";

export const LoginPage = ({startLogin}) => (

    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Boilerplate</h1>
            <p>Tag line for app</p>
            <button 
            onClick={startLogin}
            className="login-button"
            >Login with Google</button>
        </div>
    </div>
       
);

const mapDispatchToProps = (dispatch) => ({
// implicitly return an object
    startLogin: () => dispatch(startLogin())
// startLogin prop dispatches the startLogin action
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
// Component doesn't need access to anything state related  --> NO mapStateToProps
// Since I want to dispatch startLogin --> mapDispatchToProps


// If someone is already logged in --> can view private info without needing to relog in every single time