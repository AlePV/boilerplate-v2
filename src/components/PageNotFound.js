import React from "react";
import {Link} from "react-router-dom";

const PageNotFound = () => (
    <div>
        <h2>Page Not Found</h2>
        <Link to="/">Home</Link>
    </div>
);

export default PageNotFound;