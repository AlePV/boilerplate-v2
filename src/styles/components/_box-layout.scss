.box-layout {
    align-items: center;
    background: url("/images/bg.jpg");
    background-size: cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw
}
// vh --> viewport height
//   '--> 100%
// vw --> viewport width
// cover --> always covers the container with the image

.box-layout__box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: $largeSize $mediumSize;
    text-align: center;
    width: 25rem;
}


// __ --> means an element (of the layout)
// FLEX --> to center things start by setting display to flex

// PADDING : l-size m-size
//           ^^top   ^^ left&right

.box-layout__title {
    margin: 0 0 $mediumSize 0;
    line-height: 1;
}


// MARGIN --> 0    0    m-size   0
//           ^top  ^right ^bott  ^left

// LINE-HEiGHT --> space between lines