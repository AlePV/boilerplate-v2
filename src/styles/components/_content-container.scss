// MAX WIDTH CONTAINER --> to center everything
// TO GET EVERYTHING CENTERED APPROPRIATELY

.content-container {
    margin: 0 auto;
    padding: 0 $mediumSize;
    max-width: 80rem;
}


// AUTO --> evenly distributes space on left & right
//     '--> If the browser's width is 100 --> wnat to make sure 10rem are distributed in each side

// PADDING --> 0        mediumSize
//            ^top/bot    ^left&right
//        '--> makes sure content doesn't push left or right in any browser