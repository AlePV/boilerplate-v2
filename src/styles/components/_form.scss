.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $mediumSize;
    }
}

// > --> select direct descendance
// * --> looks for all children

.form__error {
    margin: 0 0 $mediumSize 0;
    font-style: italic;
}