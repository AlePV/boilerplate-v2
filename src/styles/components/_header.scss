.header {
    background: $darkBlue;
}

.header__title {
    color: white;
    text-decoration: none;
    h1 {
        margin: 0
    }
}

// TEXT_DECORATION --> clears the decoration of links (the underline)

.header__container {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $smallSize 0;
}
// Use FLEXBOX --> to distribute things appropriately how I want
// Distributes space between 
// ALIGN-ITEMS --> aligns them vertically