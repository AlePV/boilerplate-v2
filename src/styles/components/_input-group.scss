.input-group {
    // for mobile --> used inverse logic
    display: flex;
    flex-direction: column;
    margin-bottom: $mediumSize;

    // for desktop --> used inverse logic
    @media (min-width: $desktopBreakpoint) {
        flex-direction: row;
        margin-bottom: $largeSize;
    }
}

.input-group__item {
    margin-bottom: $smallSize;
    // for desktop
    @media (min-width: $desktopBreakpoint) {
        margin: 0 $smallSize 0 0
    }
}