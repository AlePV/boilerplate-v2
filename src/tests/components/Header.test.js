// Allows to render components in javascript code and assert something on what got rendered
import React from "react";
import {shallow} from "enzyme";
// import ReactShallowRenderer from "react-test-renderer/shallow";// shallow version does not include user interaction
import {Header} from "../../components/Header";


test("should render Header component correctly", () => {
    const wrapper = shallow(<Header startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();


    // expect(wrapper.find("h1").text()).toBe("Expensify");
    // const renderer = new ReactShallowRenderer;
    // // this creates a new instance of ReactShallowRenderer
    // renderer.render(<Header/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});



// returns renderedOutput of the JSX I put in 

// SNAPSHOTS --> allow to track changes to data overtime

// Can pass anything into startLogout --> so passed in an empty arrow function (that does nothing)


test("should call startLogout on button click", () => {
    const startLogout = jest.fn(); //This is a spy
    const wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find("button").simulate("click");
    expect(startLogout).toHaveBeenCalled();
});