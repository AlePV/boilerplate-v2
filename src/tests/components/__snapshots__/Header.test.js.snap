// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__container"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="login-button login-button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
